sched/scheduler/job_state.go

```diff
@@ -49,15 +50,19 @@ func (s taskStatesByDuration) Less(i, j int) bool {
 // Note: taskDurations is optional and only used to enable sorts using taskStatesByDuration above.
 func newJobState(job *sched.Job, saga *saga.Saga, taskDurations map[string]averageDuration) *jobState {
 	j := &jobState{
-		Job:            job,
-		Saga:           saga,
-		Tasks:          make([]*taskState, 0),
-		EndingSaga:     false,
-		TasksCompleted: 0,
-		TasksRunning:   0,
-		JobKilled:      false,
+		Job:               job,
+		Saga:              saga,
+		Tasks:             make([]*taskState, 0),
+		EndingSaga:        false,
+		TasksCompleted:    0,
+		TasksRunning:      0,
+		JobKilled:         false,
+		InsertionPriority: job.Def.Priority, // Shoehorn in priority to determine insertion behavior in task_scheduler.go
 	}
 
+	// Job Priorities higher than 1 are currently disabled as per stateful_scheduler.go (excepting InsertionPriority).
+	job.Def.Priority = sched.Priority(min(int(sched.P1), int(job.Def.Priority)))
```

Comment:
this is introducing a side effect on the job structure, recommend we override job.Def.Priority in thriftJobToScoot() in run_job.go

Reply:
I can't do that until we introduce a new field for Job InsertionOrder, which I'd prefer to defer for now.